clippy_lints/src/attrs/duplicated_attributes.rs

```diff
@@ -3,16 +3,18 @@ use clippy_utils::diagnostics::span_lint_and_then;
 use rustc_ast::{Attribute, MetaItem};
 use rustc_data_structures::fx::FxHashMap;
 use rustc_lint::EarlyContext;
-use rustc_span::{Span, sym};
+use rustc_span::{Span, Symbol, sym};
 use std::collections::hash_map::Entry;
 
 fn emit_if_duplicated(
     cx: &EarlyContext<'_>,
     attr: &MetaItem,
     attr_paths: &mut FxHashMap<String, Span>,
-    complete_path: String,
+    parent: &[Symbol],
+    complete_path: impl FnOnce(&str) -> String,
 ) {
-    match attr_paths.entry(complete_path) {
+    let parent = parent.iter().map(Symbol::to_string).collect::<Vec<_>>().join(":");
```

Comment:
With `Itertools`, removes the vec + strings allocations

```suggestion
    let parent = parent.iter().join(":");
```

Reply:
Neat. Done.